pytorch_blade/tests/torch-disc-pdll/utils.cpp

```diff
@@ -150,6 +151,40 @@ static LogicalResult checkTorchValueTensorLiteral(
   return success();
 }
 
+static LogicalResult checkTorchTensorElemType(
+    PatternRewriter& rewriter,
+    ArrayRef<PDLValue> values) {
+  assert(values.size() == 2);
+
+  auto v = values[0].cast<Value>();
+  auto tensorTy = v.getType().dyn_cast<Torch::ValueTensorType>();
+  if (!tensorTy)
+    return failure();
+
+  auto type_str =
+      values[1].cast<Attribute>().cast<StringAttr>().getValue().str();
+
+  std::unordered_map<std::string, Type> typeconvert_dict = {
```

Comment:
Do we just need a switch-case instead of initialize a hash map each time here?

Reply:
some if else statement? It seems that cpp could not switch with string.

Reply:
It could be a static map with value of lambda type, receiving a rewriter. Anyway, this version works.